### PR TITLE
fix: Use a Just-In-Time approach for LRECL validation

### PR DESCRIPTION
**Pull request by zFernand0**
## Proposed changes

Switch to a just--in-time checking of the LRECL.
This avoids unnecessary fetching when listing, which impacted pagination 😋 

## Release Notes

Milestone: 3.3.0

Changelog: N/A

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [x] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [x] I have tested new functionality with the FTP extension and profile verifying no extender profile type breakages introduced
- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Comment by zFernand0** [comment]
Closing in favor of: 
- https://github.com/zowe/zowe-explorer-vscode/pull/3731

Once the above PR is merged, the following issue should be resolved:
- https://github.com/zowe/zowe-explorer-vscode/pull/3791#issuecomment-3189337089

Thanks for being patient 🙏

File: packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts [hunk 1]
```diff
@@ -583,7 +583,43 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
...
+            } else {
+                entry = tempEntry as DsEntry;
+            }
+        }
+        if (dsStats) {
+            const longLines = {};
+            try {
+                const document = await vscode.workspace.openTextDocument(uri);
+                for (let i = 0; i < document.lineCount; i++) {
+                    if (document.lineAt(i).text.length > (Number(dsStats.lrecl) || Number(dsStats.blksz))) {
```
Comment (traeok) [inline packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts:608]: Thoughts on skipping this logic if `lrecl` and `blksz` are both `undefined` in `dsStats`?

According to the z/OSMF list data sets response w/ `X-IBM-Attributes=base`, neither `lrecl` and `blksz` are required properties in the `items` array:

https://www.ibm.com/docs/en/zos/2.5.0?topic=zdsfri-json-document-specifications-zos-data-set-file-rest-interface-requests#RESTFILES_JSONDocumentSpecifications__dataSetListWithAttributes

If neither are included with the response, this logic would still run for every line in the document, which feels unnecessary.

**Review by JillieBeanSim — APPROVED** [review]
Thanks so much @zFernand0 for this fix ❤️

File: packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts (hunk 1 above)
Comment (zFernand0) [reply packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts:608]: Great observation! 🙏 

Resolved in 4898ac8fbe5fcf124f10c63c91638d3ecb7d60bd
Plus added a test just in case 🙏 

Good catch 🙏

**Comment by zFernand0** [comment]
Thanks for the quick review, Billie! 🙏 

In the final commit, I just addressed the edge case in which no lrecl nor blksz is returned.
Plus added a small unit test for it! 🙏 

I don't believe it requires a re-review since the final commit is mostly to avoid going through the entire document unnecessarily (if we don't have enough information) 😋

**Comment by zFernand0** [comment]
For awareness 🙏 

It seems like we have a transient unit test error on node18-macos...
https://github.com/zowe/zowe-explorer-vscode/actions/runs/17071658014/job/48401936307?pr=3810#step:7:183
image

not sure if it's just too fast for GH Actions 😅 

Running locally with node 18 on macos (or a simple replay) just worked 🙏
